### ISSUE TITLE
Block release on app version consistency

### ISSUE THREAD

**Issue by absoludity** [kubeapps/kubeapps#3143]
### Description:

The Bitnami internal infrastructure which is used to release new Kubeapps chart versions currently watches a "main" image (for us, kubeapps-dashboard). When that image is updated, a new chart release with updated versions is created, tested and published.

It turns out that this can release a chart version with inconsistent app versions across Kubeapps components, as long as the chart tests pass. For example:

```
$ helm fetch bitnami/kubeapps --version 7.1.3 --untar && grep "repository: bitnami/kubeapps-" -A1 kubeapps/values.yaml
    repository: bitnami/kubeapps-dashboard
    tag: 2.3.3-debian-10-r0
--
    repository: bitnami/kubeapps-apprepository-controller
    tag: 2.3.3-scratch-r0
--
    repository: bitnami/kubeapps-asset-syncer
    tag: 2.3.3-scratch-r0
--
    repository: bitnami/kubeapps-kubeops
    tag: 2.3.3-scratch-r0
--
    repository: bitnami/kubeapps-assetsvc
    tag: 2.3.3-scratch-r0
--
    repository: bitnami/kubeapps-pinniped-proxy
    tag: 2.3.2-debian-10-r60
```

If we want to ensure that this is not possible, we should add a test that ensures all app version tags are the same. If we want to ensure it's not possible to release a new chart minor version without a manual check, we should add a test that ensures all version tags match a specific minor version (not nice, but effectively enables us to require a PR to do the release). Thoughts?

**Comment by antgamdia** [comment]
I totally agree with you, a test has to be added to avoid that kind of unfortunate situations that prevented us from a smooth release process.
However, I have a couple of questions:

>  currently watches a "main" image (for us, kubeapps-dashboard)

1. Are you 100% sure? I assumed it was trigger when a new tag was detected at https://github.com/kubeapps/kubeapps/tags 

2. Even if we add the test as part of our test suite, how this would prevent Bitnami from releasing an inconsistent version? (I mean, they have a different test suite totally unsynced with ours)

**Comment by absoludity** [comment]
> I totally agree with you, a test has to be added to avoid that kind of unfortunate situations that prevented us from a smooth release process.
> However, I have a couple of questions:
> 
> > currently watches a "main" image (for us, kubeapps-dashboard)
> 
>     1. Are you 100% sure?

Just based off Juan's comment:

> Please note it mainly tracks the main component (kubeapps-dashboard in this case). So.. When the release of the main component has been completed in DockerHub, it will trigger the chart-pr...

And it matched the fact that another auto release was performed ([7.1.6](https://github.com/bitnami/charts/commit/18697f9ab023379b5d5dad4e4c92e2f86a3a95ef#diff-6d9a07490bcfd3474aa256c98627b62c50ae0753a3764ca33d2bf9c682a0ecb6)) when the pipeline updated the kubeapps-dashboard image due to a new nginx version.

> I assumed it was trigger when a new tag was detected at https://github.com/kubeapps/kubeapps/tags

Right, that triggers building and publishing the new bitnami images for Kubeapps (including the dashboard). But I'm assuming, based on the info that we have, that the chart release process is loosely coupled to the publishing of new bitnami images, in that the chart release process is triggered when it notices that the kubeapps-dashboard image has been updated (kubeapps-dashboard being chosen as the "main" image for the kubeapps app). This is why we see new chart releases independent of us tagging a new image (see 7.1.6 above). 

> 
>     2. Even if we add the test as part of our test suite, how this would prevent Bitnami from releasing an inconsistent version? (I mean, they have a different test suite totally unsynced with ours)

Yes, absolutely. The test needs to be in the pipeline. I created this issue only so that we can discuss what might be the best way for us to ensure that we can still control the release of a new chart. The original suggestion of all images being the same versions would avoid the issue of inconsistent versions, but not necessarily avoid releasing a new chart when we're not ready (ie. we have other chart updates we want included).
